### PR TITLE
Frogbot shouldn't attempt upgrading Go and Pip

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---
- Frogbot should not try to fix vulnerabilities in Go itself or pip/setuptools/wheel, because they cannot be fixed through their package descriptors, and only can be upgraded in the environment itself.

### CODE REVIEW

File: commands/utils/packagehandlers/pythonpackagehandler.go [hunk 1]
```diff
@@ -35,7 +44,7 @@ func (py *PythonPackageHandler) UpdateImpactedPackage(impactedPackage string, fi
 	case coreutils.Pipenv:
 		return py.GenericPackageHandler.UpdateImpactedPackage(impactedPackage, fixVersionInfo, extraArgs...)
 	default:
-		return errors.New("Unknown python package manger: " + fixVersionInfo.PackageType.GetPackageType())
+		return fmt.Errorf("unknown python package manger: %s", fixVersionInfo.PackageType.GetPackageType())
```
Comment (sverdlov93) [inline commands/utils/packagehandlers/pythonpackagehandler.go:47]:
```suggestion
		return errors.New("unknown python package manger: " + fixVersionInfo.PackageType.GetPackageType())
```

File: commands/utils/packagehandlers/gopackagehandler.go [hunk 2]
```diff
@@ -2,14 +2,21 @@ package packagehandlers
...
+	GoPackage = "github.com/golang/go"
 )
 
 type GoPackageHandler struct {
 	GenericPackageHandler
 }
 
 func (golang *GoPackageHandler) UpdateImpactedPackage(impactedPackage string, fixVersionInfo *utils.FixVersionInfo, extraArgs ...string) error {
-	impactedPackage = strings.Trim(impactedPackage, "v")
+	if impactedPackage == GoPackage {
```
Comment (sverdlov93) [inline commands/utils/packagehandlers/gopackagehandler.go:17]: where did `impactedPackage = strings.Trim(impactedPackage, "v")` disappear?

File: commands/utils/packagehandlers/gopackagehandler.go [hunk 3]
```diff
@@ -2,14 +2,21 @@ package packagehandlers
 
 import (
 	"github.com/jfrog/frogbot/commands/utils"
-	"strings"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+)
+
+const (
+	GoPackage = "github.com/golang/go"
```
Comment (sverdlov93) [inline commands/utils/packagehandlers/gopackagehandler.go:9]:
```suggestion
	goPackage = "github.com/golang/go"
```

File: commands/utils/packagehandlers/gopackagehandler.go [hunk 4]
```diff
@@ -2,14 +2,21 @@ package packagehandlers
...
 )
 
 type GoPackageHandler struct {
 	GenericPackageHandler
 }
 
 func (golang *GoPackageHandler) UpdateImpactedPackage(impactedPackage string, fixVersionInfo *utils.FixVersionInfo, extraArgs ...string) error {
-	impactedPackage = strings.Trim(impactedPackage, "v")
+	if impactedPackage == GoPackage {
+		log.Info("Skipping vulnerable package", impactedPackage, "since it is not defined in your go.mod file. Update Go version to", fixVersionInfo.FixVersion, "to fix this vulnerability.")
```
Comment (sverdlov93) [inline commands/utils/packagehandlers/gopackagehandler.go:18]: duplicate code with python.

File: commands/createfixpullrequests_test.go [hunk 5]
```diff
@@ -148,11 +153,13 @@ func TestFixPackageVersion(t *testing.T) {
 				// Fix impacted package for each technology
 				fixVersionInfo := utils.NewFixVersionInfo(test.fixVersion, test.technology, true)
 				assert.NoError(t, cfg.updatePackageToFixedVersion(test.impactedPackaged, fixVersionInfo))
-				file, err := os.ReadFile(test.packageDescriptor)
-				assert.NoError(t, err)
-				assert.Contains(t, string(file), test.fixVersion)
-				// Verify that case-sensitive packages in python are lowered
-				assert.Contains(t, string(file), strings.ToLower(test.impactedPackaged))
+				if !test.shouldNotFix {
```
Comment (sverdlov93) [inline commands/createfixpullrequests_test.go:156]: that test checks that all shouldFix deps are fixed,
but it doesn't check that all shouldnotFix deps are indeed not fixed

File: commands/utils/packagehandlers/pythonpackagehandler.go (hunk 1 above)
Comment (omerzi) [reply commands/utils/packagehandlers/pythonpackagehandler.go:47]: done

File: commands/utils/packagehandlers/gopackagehandler.go (hunk 2 above)
Comment (omerzi) [reply commands/utils/packagehandlers/gopackagehandler.go:17]: The impacted package is the vulnerable package name, we should not trim its name. Also, the fix version is coming from Xray without the "v" prefix of Go, so it's unnecessary.

File: commands/utils/packagehandlers/gopackagehandler.go (hunk 3 above)
Comment (omerzi) [reply commands/utils/packagehandlers/gopackagehandler.go:9]: done

File: commands/utils/packagehandlers/gopackagehandler.go (hunk 4 above)
Comment (omerzi) [reply commands/utils/packagehandlers/gopackagehandler.go:18]: Agree

File: commands/createfixpullrequests_test.go (hunk 5 above)
Comment (omerzi) [reply commands/createfixpullrequests_test.go]: done

**Review by sverdlov93 — APPROVED** [review]

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: commands/createfixpullrequests.go [hunk 6]
```diff
@@ -305,6 +306,12 @@ func (cfp *CreateFixPullRequestsCmd) updatePackageToFixedVersion(impactedPackage
 			}
 		}()
 	}
+	// Skip build tools dependencies (for example: pip) that are not defined by the descriptor file and cannot be fixed by PR.
```
[the remark this answers is not in the document]
Comment (omerzi) [reply commands/createfixpullrequests.go]: done

**Pull request merged**